Assets/BowAndArrowImproved/Scripts/Leaderboard/Leaderboard.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class Leaderboard : MonoBehaviour
+{
+    private const string JsonLeaderBoardFileName = "JsonLeaderboardHolder.json";
+    private LeaderboardEntries _leaderboardEntries;
+
+    private IDataLoader _leaderboardLoader = new JsonDataLoader();
+    private IDataSaver _leaderboardSaver = new JsonDataSaver();
+
+    [SerializeField] private Scoreboard scoreboard;
+    [SerializeField] private GameObject scoreEntriesGrid;
+    [SerializeField] private ScoreEntry scoreEntryPrefab;
+    [SerializeField] private int numOfEntries = 10;
+
+    private List<ScoreEntry> _scoresEntries = new List<ScoreEntry>();
+
+    private int _myScoreEntryPosition = -1;
+    private int _myScore = 0;
+
+    public void OnGameManagerStateChanged(GameManager.State newState)
+    {
+        if (newState == GameManager.State.Lose)
+        {
+            LoadLeaderboardData();
+
+            _myScore = scoreboard._score;
+            SubmitScore();
+
+            GenerateLeaderboardEntries();
+            gameObject.SetActive(true);
```

Comment:
extract this to a different method called `HandleLoseState` or something like that, so it'll be easier to understand what's going on here and add more code to this event function in the future